### PR TITLE
Fix field acccess interval tests

### PR DESCRIPTION
**Pull request by twicki**
## Technical Description

Rework the testing framework. Changes the field_access_interval test to only use IIR and not go through SIR.

### Resolves / Enhances

Work on https://github.com/MeteoSwiss-APN/dawn/issues/702

### CODE REVIEW

**Comment by twicki** [comment]
launch jenkins

File: dawn/test/unit-test/dawn/Optimizer/Passes/TestFieldAccessIntervals.cpp [hunk 1]
```diff
@@ -12,191 +12,62 @@
...
 
-#include "dawn/Compiler/DawnCompiler.h"
-#include "dawn/Compiler/Options.h"
+#include "dawn/IIR/IIR.h"
 #include "dawn/IIR/StencilInstantiation.h"
-#include "dawn/SIR/SIR.h"
-#include "dawn/Serialization/SIRSerializer.h"
+#include "dawn/Serialization/IIRSerializer.h"
 #include "test/unit-test/dawn/Optimizer/TestEnvironment.h"
 #include <fstream>
```
Comment (jdahm) [inline dawn/test/unit-test/dawn/Optimizer/Passes/TestFieldAccessIntervals.cpp]: Does this need fstream?

Comment (twicki) [reply dawn/test/unit-test/dawn/Optimizer/Passes/TestFieldAccessIntervals.cpp]: good catch

**Comment by twicki** [comment]
launch jenkins

**Pull request merged**